endava-apprentice/src/main/java/com/example/endavaapprentice/Service/EventCompositeService.java

```diff
@@ -33,6 +39,37 @@ public Event fetchOneEvent(Long eventID) {
         return this.eventService.fetchOneEvent(eventID);
     }
 
+    @Override
+    public EventVenueEventTypeDTO fetchEventByVenueAndEventType(Long venueID, String eventType) {
+        List<Event> eventList = this.eventService.fetchAllEvents();
+        List<TicketCategoryDTO> ticketCategoryDTOS = new ArrayList<>();
+        for(Event event: eventList){
```

Comment:
this is not the prettiest nor the most efficient way of doing things. I would suggest to use at least a stream() from Java 8, to avoid inner looping and this long code. with the stream, you can first filter based on venueId and eventType, and you will keep in your list only matching events. also, you can split this into multiple methods, to create the dtos, and obtain the list of category tickets